### PR TITLE
Sync reset reenable

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
## Submitter Checklist:

Fixes https://github.com/brave/brave-browser/issues/3190.

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues/3190#issuecomment-514802188) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
```
[237/237] BraveRewardsBrowserTest.ProcessPendingContributions (TIMED OUT)
1 test timed out:
    BraveRewardsBrowserTest.ProcessPendingContributions (../../brave/components/brave_rewards/browser/rewards_service_browsertest.cc:2150)
```
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
Use STR from https://github.com/brave/brave-browser/issues/3190#issuecomment-514802188

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_sync/brave_sync_service_impl.cc [hunk 1]
```diff
@@ -132,7 +132,7 @@ BraveSyncClient* BraveSyncServiceImpl::GetSyncClient() {
 }
 
 bool BraveSyncServiceImpl::IsSyncConfigured() {
-  return sync_configured_;
+  return sync_configured_ && (sync_prefs_->GetSyncDevices()->size() >= 2);
```
Comment (darkdh) [inline components/brave_sync/brave_sync_service_impl.cc]: can you elaborate why we need this change? we will reset sync chain when containing only one device https://github.com/brave/brave-core/blob/ede3ca626e86a5d60ba12786b0e74c91870e2acd/components/brave_sync/brave_sync_service_impl.cc#L500

Comment (AlexeyBarabash) [reply components/brave_sync/brave_sync_service_impl.cc]: @darkdh 
In the situation described in the issue , https://github.com/brave/brave-browser/issues/3190#issuecomment-514802188 , when second `OnSetupSyncHaveCode/OnSetupSyncNewToSync` happens , there is already one device in the devices list, so without patch above execution will not reach `ForceCompleteReset();` in `OnSetupSyncHaveCode/OnSetupSyncNewToSync` .

The more clear way would be to keep `IsSyncConfigured()` as it was and modify `OnSetupSyncHaveCode/OnSetupSyncNewToSync` to 

```
  if (IsSyncConfigured() && (sync_prefs_->GetSyncDevices()->size() >= 2)) {
    NotifyLogMessage("already configured");
    return;
  }
```

Comment (darkdh) [reply components/brave_sync/brave_sync_service_impl.cc]: it is ok to leave it in `IsSyncConfigured`, can you put some comments in it?

Comment (AlexeyBarabash) [reply components/brave_sync/brave_sync_service_impl.cc]: thanks, fixed at
https://github.com/brave/brave-core/pull/3038/files#diff-c04ecef4235e311e0a383fe4fabcb4d4R135

**Comment by AlexeyBarabash** [comment]
@darkdh could you please re-approve, as I had to make force-push to deal with CI

**Pull request merged**